tests/integrationv2/processes.py

```diff
@@ -182,7 +193,10 @@ def _communicate(self, input_data=None, send_marker_list=None, close_marker=None
                         # If we are looking for, and find, the ready-to-send marker, then
                         # register STDIN to receive events. If there is no data to send,
                         # just mark input_send as true so we can close out STDIN.
-                        if send_marker is not None and send_marker in str(data):
+                        if send_marker:
+                            print(f'{self.name}: looking for send_marker {send_marker} in {data_debug}')
+                        if send_marker is not None and send_marker in data_str:
```

Comment:
Can't `data_debug` and `data_str` be undefined here?

Reply:
Oh good catch! Yes, and python doesn't actually care if you do str(None), and that's what the code did before, so let's just always set data_str and data_debug.